enhancements/image-references.md

```diff
@@ -0,0 +1,230 @@
+---
+title: image-references-in-operator-bundles
+authors:
+  - "@stevekuznetsov"
+reviewers:
+  - "@jwforres"
+  - "@shawn-hurley"
+  - "@gallettilance"
+  - "@lcarva"
+approvers:
+  - "@jwforres"
+  - "@shawn-hurley"
+creation-date: 2020-05-17
+last-updated: 2020-05-17
+status: implementable
+see-also:
+  - "/enhancements/olm/operator-bundle.md"
+---
+
+# Image References in Operator Bundle Manifests
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Manifests that make up an Operator Bundle for installation via the Operator
+Lifecycle Manager refer to one or more container images by their pull
+specifications: these container images define the operator and operands that
+the manifests deploy and manage. As with OpenShift release payloads, operator
+bundles must refer to images by digest in order to produce reproducible
+installations. A shared process to build operator bundles that replaces image
+references with fully-resolved pull specifications that pin images by digest
+must be built; this process must allow for a number of separate build systems
+to direct how these replacements occur in order to support a full-featured
+build and test strategy.
+
+## Motivation
+
+### Goals
+
+- there is one, canonical, method for building an operator bundle images
+  from a directory of manifests
+- it is possible to perform image resolution and pinning separate from bundle
+  creation, but not the opposite
+- building an operator bundle image does not require the use of a container
+  runtime, elevated privileges or any capacities that are not present for 
+  containerized workloads on OpenShift
+- it is as simple for a developer to build a bundle referring to test versions
+  of operand images as it is for a CI or productized build pipleine to create
+  bundle images for publication
+- operator manifest authors dictate the set of image references in manfiests
+  that must be resolved and pinned
+- operator bundle images may be inspected to determine the pull specifications
+  that were used in the creation of the bundle
+
+### Non-Goals
+
+- operator manifest authors must not be required to define the registry from
+  which any individual build system will resolve image references
+- upstream operator manifests must not be required to know how common names or
```

Comment:
Same here, either this is a goal in disguise or poorly worded. If this is actually not a goal, it should be something like
```
* Requiring upstream operator manifests to know how common names or references change when built in a downstream pipeline
```

Reply:
They should not be involved in the "how" of resolution in _any_ part of the build system.

Reply:
okay so it is a goal? `Must` and `Must not` can never describe the absence of a goal (aka `non-goal`). If something should explicitly be avoided, thats a goal, not a non-goal.